File: GXpert/GXpert.Web/Modules/ServerTypes/Institute/InstituteClassForm.ts
import { IntegerEditor, LookupEditor, StringEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";

export interface InstituteClassForm {
    InstituteId: IntegerEditor;
    DepartmentId: IntegerEditor;
    CourseId: LookupEditor;
    ClassId: LookupEditor;
    SemesterId: IntegerEditor;
    Title: StringEditor;
    Division: StringEditor;
    ClassTeacherId: LookupEditor;
    AcademicYearId: LookupEditor;
}

export class InstituteClassForm extends PrefixedContext {
    static readonly formKey = 'Institute.InstituteClass';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!InstituteClassForm.init)  {
            InstituteClassForm.init = true;

            var w0 = IntegerEditor;
            var w1 = LookupEditor;
            var w2 = StringEditor;

            initFormType(InstituteClassForm, [
                'InstituteId', w0,
                'DepartmentId', w0,
                'CourseId', w1,
                'ClassId', w1,
                'SemesterId', w0,
                'Title', w2,
                'Division', w2,
                'ClassTeacherId', w1,
                'AcademicYearId', w1
            ]);
        }
    }
}

File: GXpert/GXpert.Web/Modules/Institute/InstituteClass/InstituteClassForm.cs
using Serenity.ComponentModel;
using System;

namespace GXpert.Institute.Forms;

[FormScript("Institute.InstituteClass")]
[BasedOnRow(typeof(InstituteClassRow), CheckNames = true)]
public class InstituteClassForm
{
    public int InstituteId { get; set; }
    [HalfWidth]
    public int DepartmentId { get; set; }
    [HalfWidth]
    public int CourseId { get; set; }

    [HalfWidth]
    public int ClassId { get; set; }

    
    public int SemesterId { get; set; }
    [HalfWidth]
    public string Title { get; set; }
    [HalfWidth]
    
    public string Division { get; set; }
    [HalfWidth]
    
    public int ClassTeacherId { get; set; }
        
    [HalfWidth]
    public int AcademicYearId { get; set; }
  
   
}